### PR TITLE
Rds default user name

### PR DESCRIPTION
**Pull request by fatima99s**

### CODE REVIEW

File: plugins/aws/rds/rdsDefaultUserName.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
+var async = require('async');
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'RDS Instance Should Not Have Default Username',
```
Comment (alphadev4) [inline plugins/aws/rds/rdsDefaultUserName.js:5]:
```suggestion
    title: 'RDS Default Username',
```

File: plugins/aws/rds/rdsDefaultUserName.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
+var async = require('async');
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'RDS Instance Default Master Username',
+    category: 'RDS',
+    domain: 'Databases',
+    description: 'Ensures RDS instance doest not have a default master username.',
```
Comment (alphadev4) [inline plugins/aws/rds/rdsDefaultUserName.js:8]:
```suggestion
    description: 'Ensures RDS instance does not have default master username.',
```

File: plugins/aws/rds/rdsDefaultUserName.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+            if (!describeDBInstances.data.length) {
+                helpers.addResult(results, 0, 'No RDS instances found', region);
+                return rcb();
+            }
+
+            var defaultusername=["admin","postgres"];
+
+            describeDBInstances.data.forEach(instance=>{
+                if (!instance.DBInstanceArn||!instance.MasterUsername) return;
```
Comment (alphadev4) [inline plugins/aws/rds/rdsDefaultUserName.js:39]:
```suggestion
                if (!instance.DBInstanceArn || !instance.MasterUsername) return;
```

File: plugins/aws/rds/rdsDefaultUserName.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+                return rcb();
+            }
+
+            var defaultusername=["admin","postgres"];
+
+            describeDBInstances.data.forEach(instance=>{
+                if (!instance.DBInstanceArn || !instance.MasterUsername) return;
+
+                if (defaultusername.includes(instance.MasterUsername)) {
+                    helpers.addResult(results, 2, 'RDS instance have a default master username',
```
Comment (alphadev4) [inline plugins/aws/rds/rdsDefaultUserName.js:42]:
```suggestion
                    helpers.addResult(results, 2, 'RDS instance has a default master username',
```

File: plugins/aws/rds/rdsDefaultUserName.spec.js [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+                done();
+             });
+        });
+     
+       it('should fail if the username is default of the rds instace',function(done){
+           const cache=createCache([describeDBInstances[0]]);
+           rdsDefaultUserName.run(cache,{},(err,results)=>{
+                expect(results.length).to.equal(1);
+                expect(results[0].status).to.equal(2);
+                expect(results[0].message).includes('RDS instance have a default master username');
```
Comment (alphadev4) [inline plugins/aws/rds/rdsDefaultUserName.spec.js:100]:
```suggestion
                expect(results[0].message).includes('RDS instance has a default master username');
```

File: plugins/aws/rds/rdsDefaultUserName.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+            var defaultusername=["admin","postgres"];
+
+            describeDBInstances.data.forEach(instance=>{
+                if (!instance.DBInstanceArn || !instance.MasterUsername) return;
+
+                if (defaultusername.includes(instance.MasterUsername)) {
+                    helpers.addResult(results, 2, 'RDS instance has a default master username',
+                    region, instance.DBInstanceArn);
+                }else{
```
Comment (alphadev4) [inline plugins/aws/rds/rdsDefaultUserName.js:44]:
```suggestion
                } else {
```

File: plugins/aws/rds/rdsDefaultUserName.js [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+            }
+
+            if (!describeDBInstances.data.length) {
+                helpers.addResult(results, 0, 'No RDS instances found', region);
+                return rcb();
+            }
+
+            var defaultusername=["admin","postgres"];
+
+            describeDBInstances.data.forEach(instance=>{
```
Comment (alphadev4) [inline plugins/aws/rds/rdsDefaultUserName.js:38]:
```suggestion
            describeDBInstances.data.forEach(instance => {
```

File: plugins/aws/rds/rdsDefaultUserName.js [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+            }
+
+            var defaultusername=["admin","postgres"];
+
+            describeDBInstances.data.forEach(instance => {
+                if (!instance.DBInstanceArn || !instance.MasterUsername) return;
+
+                if (defaultusername.includes(instance.MasterUsername)) {
+                    helpers.addResult(results, 2, 'RDS instance has a default master username',
+                    region, instance.DBInstanceArn);
```
Comment (alphadev4) [inline plugins/aws/rds/rdsDefaultUserName.js:43]:
```suggestion
                        region, instance.DBInstanceArn);
```

File: plugins/aws/rds/rdsDefaultUserName.js [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+                } else {
+                    helpers.addResult(results, 0, 'RDS instance does not have a default master username',
+                    region, instance.DBInstanceArn);
```
Comment (alphadev4) [inline plugins/aws/rds/rdsDefaultUserName.js:46]:
```suggestion
                        region, instance.DBInstanceArn);
```

**Review by mehakseedat63 — APPROVED** [review]

**Review by mehakseedat63 — CHANGES_REQUESTED** [review]
Resolve merge conflicts

**Pull request closed without merging**